cli/tests/integration/invitations/shared_recovery.rs

```diff
@@ -0,0 +1,204 @@
+            ParsecInvitationAddr::new(
+                alice.organization_addr.clone(),
+                alice.organization_id().clone(),
+                InvitationType::ShamirRecovery,
+                token,
+            )
+        }
+        rep => {
+            panic!("Server refused to create user invitation: {rep:?}");
+        }
+    };
+
+    let token = invitation_addr.token();
+
+    // spawn greeter thread
+    let mut cmd_greeter = assert_cmd::Command::cargo_bin("parsec-cli").unwrap();
+    cmd_greeter.args([
+        "invite",
+        "greet",
+        "--device",
+        &bob.device_id.hex(),
+        &token.hex().to_string(),
+    ]);
+
+    let program_greeter = cmd_greeter.get_program().to_str().unwrap().to_string();
+    let program_greeter = cmd_greeter
+        .get_args()
+        .fold(program_greeter, |acc, s| format!("{acc} {s:?}"));
+
+    let p_greeter = Arc::new(Mutex::new(
+        spawn(&dbg!(program_greeter), Some(1000)).unwrap(),
+    ));
+
+    // spawn claimer thread
+
+    let mut cmd_claimer = assert_cmd::Command::cargo_bin("parsec-cli").unwrap();
+    cmd_claimer.args([
+        "invite",
+        "claim",
+        invitation_addr.to_url().as_ref(),
+        "--password-stdin",
+    ]);
+
+    let program_claimer = cmd_claimer.get_program().to_str().unwrap().to_string();
+    let program_claimer = cmd_claimer
+        .get_args()
+        .fold(program_claimer, |acc, s| format!("{acc} {s:?}"));
+
+    let p_claimer = Arc::new(Mutex::new(
+        spawn(&dbg!(program_claimer), Some(10_000)).unwrap(),
+    ));
+
+    // retrieve greeter code
+    let greeter_cloned = p_greeter.clone();
+    let greeter = tokio::task::spawn(async move {
+        let mut locked = greeter_cloned.lock().unwrap();
+
+        locked.exp_string("Enter password for the device:").unwrap();
+        locked.send_line(DEFAULT_DEVICE_PASSWORD).unwrap();
+        locked.exp_string("Waiting for claimer").unwrap();
+    });
+    let claimer_cloned = p_claimer.clone();
+
+    let claimer = tokio::task::spawn(async move {
+        let mut locked = claimer_cloned.lock().unwrap();
+        locked
+            .exp_string("Choose a person to contact now:")
+            .unwrap();
+        // down to choose bob
+        locked.send_line("j").unwrap();
+
+        //  p_claimer.exp_string(&format!("Waiting the greeter {} to start the invitation procedure", bob.human_handle)).unwrap();
+        locked
+            .exp_string("Select code provided by greeter:")
+            .unwrap();
+    });
+    greeter.await.unwrap();
+    p_greeter
+        .lock()
+        .unwrap()
+        .exp_string("Code to provide to claimer:")
+        .unwrap();
+    let (_, matched) = p_greeter.lock().unwrap().exp_regex("[A-Z0-9]{4}").unwrap();
+    let sas_code = dbg!(matched[matched.len() - 4..].to_string()); // last 4 chars are the sas code
+
+    // code selection
+
+    claimer.await.unwrap();
+    let cloned_claimer = p_claimer.clone();
```

Comment:
Why do you need to clone here?

Reply:
Because I move that into the block just below, and I need to keep access for the last part.

And I'd rather lock the greeter and claimer sessions in their own scope to avoid confusion.